apps/api-journeys/src/app/modules/journey/journey.resolver.ts

```diff
@@ -160,6 +182,65 @@ export class JourneyResolver {
     })
   }
 
+  @Mutation()
+  @UseGuards(RoleGuard('id', UserJourneyRole.owner))
+  async journeyArchive(@Args('id') id: string): Promise<Journey> {
+    return await this.journeyService.update(id, {
+      status: JourneyStatus.archived,
+      archivedAt: new Date().toISOString()
+    })
+  }
+
+  @Mutation()
+  @UseGuards(RoleGuard('id', UserJourneyRole.owner))
+  async journeyDelete(@Args('id') id: string): Promise<Journey> {
+    return await this.journeyService.update(id, {
+      status: JourneyStatus.deleted,
+      deletedAt: new Date().toISOString()
+    })
+  }
+
+  @Mutation()
+  @UseGuards(RoleGuard('id', UserJourneyRole.owner))
+  async journeyTrash(@Args('id') id: string): Promise<Journey> {
+    return await this.journeyService.update(id, {
+      status: JourneyStatus.trashed,
+      trashedAt: new Date().toISOString()
+    })
+  }
+
+  async journeyTrashAllArchived(
+    @CurrentUserId() userId: string
+  ): Promise<Journey[]> {
+    const results: Journey[] = (await this.journeyService.getAllByOwner(userId))
+      .filter((j) => j.status === JourneyStatus.archived)
+      .map((j) => ({
+        ...j,
+        status: JourneyStatus.deleted,
+        deletedAt: new Date().toISOString()
+      }))
+
+    return await this.journeyService.updateAll(results)
+  }
+
+  @Mutation()
+  @UseGuards(RoleGuard('id', UserJourneyRole.owner))
+  async journeyRestore(@Args('id') id: string): Promise<Journey> {
+    const result: Journey = await this.journeyService.get(id)
+    const lastActiveStatus =
+      result.publishedAt == null ? JourneyStatus.draft : JourneyStatus.published
+
+    if (
+      result.status === JourneyStatus.draft ||
+      result.status === JourneyStatus.published
+    ) {
+      throw new UserInputError('Journey is already active')
+    }
+    return await this.journeyService.update(id, {
+      status: lastActiveStatus
+    })
```

Comment:
small nit. not worth calculating lastActiveStatus unless the if statement is false.

```suggestion
    if (
      result.status === JourneyStatus.draft ||
      result.status === JourneyStatus.published
    ) {
      throw new UserInputError('Journey is already active')
    }
    
    const lastActiveStatus =
      result.publishedAt == null ? JourneyStatus.draft : JourneyStatus.published

    return await this.journeyService.update(id, {
      status: lastActiveStatus
    })
```